### PR TITLE
Update deployment scripts

### PR DESCRIPTION
**Pull request by darrenj**
- Deployment scripts were out of step between VSIX and Generator so updating with the latest.

### CODE REVIEW

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1 [hunk 1]
```diff
@@ -47,48 +90,74 @@ if (-not $appId) {
...
 # Deploy Azure services (deploys LUIS, QnA Maker, Content Moderator, CosmosDB)
-Write-Host "Deploying Azure services ..."
+Write-Host "> Deploying Azure services (this could take a while)..." -ForegroundColor Yellow
 if ($parametersFile) {
-    az group deployment create `
+    (az group deployment create `
         --name $timestamp `
         --resource-group $resourceGroup `
-        --template-file "$(Join-Path $PSScriptRoot '..' 'resources' 'template.json')" `
+        --template-file "$(Join-Path $PSScriptRoot '..' 'Resources' 'template.json')" `
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1:102]: The `Resources` folder should be in lowercase.

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1 [hunk 2]
```diff
@@ -47,48 +90,74 @@ if (-not $appId) {
...
+        --parameters microsoftAppId=$appId microsoftAppPassword="`"$($appPassword)`"") 2>> $logFile | Out-Null
 }
 else {
-    az group deployment create `
+    (az group deployment create `
         --name $timestamp `
         --resource-group $resourceGroup `
-        --template-file "$(Join-Path $PSScriptRoot '..' 'resources' 'template.json')" `
-        --parameters microsoftAppId=$appId microsoftAppPassword=$appPassword | Out-Null
+        --template-file "$(Join-Path $PSScriptRoot '..' 'Resources' 'template.json')" `
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1:110]: The `Resources` folder should be in lowercase.

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1 [hunk 3]
```diff
@@ -1,44 +1,87 @@
...
+	[string] $resourceGroup,
     [string] $location,
 	[string] $appId,
     [string] $appPassword,
     [string] $luisAuthoringKey,
+	[string] $luisAuthoringRegion,
     [string] $parametersFile,
-	[string] $outFolder = $(Join-Path $(Get-Location) src)
+	[string] $languages = "en-us",
+	[string] $outFolder = $(Get-Location),
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1:13]: `$outFolder` should point to the `./src` folder using `[string] $outFolder = $(Join-Path $(Get-Location) src)`

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1 [hunk 4]
```diff
@@ -1,119 +1,190 @@
...
+Write-Host "> Deploying cognitive models ..."
+foreach ($language in $languageArr)
 {
     $langCode = ($language -split "-")[0]
 
     $config = @{
         dispatchModel = New-Object PSObject
         languageModels = @()
-        knowledgeBases = @()
+        knowledgebases = @()
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1:78]: This should use correct casing for TypeScript (`knowledgeBases`)

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1 [hunk 5]
```diff
@@ -1,119 +1,190 @@
...
-            version = $luisApp.activeVersion
-            region = $location
-        }
+			# Add to config 
+			$config.languageModels += @{
+				id = $lu.BaseName
+				name = $luisApp.name
+				appid = $luisApp.id
+				authoringkey = $luisauthoringkey
+				subscriptionkey = $luisauthoringkey
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1:115]: This values should be `appId`, `authoringKey`, and `subscriptionKey`

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1 [hunk 6]
```diff
@@ -1,119 +1,190 @@
...
+	else {
+		$dispatchApp  = $dispatch | ConvertFrom-Json
+
+	    # Add to config
+		$config.dispatchModel = @{
+			type = "dispatch"
+			name = $dispatchApp.name
+			appid = $dispatchApp.appId
+			authoringkey = $luisauthoringkey
+			subscriptionkey = $luisauthoringkey
```
Comment (enzocano) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy_cognitive_models.ps1:180]: This values should be `appId`, `authoringKey`, and `subscriptionKey` (TypeScript is case-sensitive for .json files)

**Review by lzc850612 — APPROVED** [review]

**Review by enzocano — CHANGES_REQUESTED** [review]
It seems that this update breaks the case of some properties that we are using in TypeScript.
If you want, we can create other PR with the parity of the deployment scripts.

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1 (hunk 1 above)
Comment (darrenj) [reply templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1:102]: are you on a mac/osx?

Comment (darrenj) [reply templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/customAssistant/deployment/scripts/deploy.ps1:102]: I think we an go with making lowercase and we can fix on the c# side in the next release as the case sensitivity isn't an issue on windows

**Pull request merged**